app/cdap/components/RecipeDetails/index.tsx

```diff
@@ -0,0 +1,150 @@
+import { dateFormatting } from 'components/RecipeDetails/utils';
+
+interface IRecipeDetailsProps {
+  recipeDetails: IRecipeItem;
+  onCloseDetail: () => void;
+}
+
+const PREFIX = 'features.WranglerNewUI.RecipeDetails';
+
+const StyledDrawer = styled(Drawer)`
+  & .MuiDrawer-paper {
+    top: 46px;
+    height: calc(100vh - 47px);
+  }
+`;
+
+const DrawerContainerBox = styled(Container)`
+  width: 460px;
+  height: 100%;
+  padding-left: 30px;
+`;
+
+const RecipeDetailWrapper = styled(Box)`
+  margin-top: 20px;
+`;
+
+const RecipeName = styled(Typography)`
+  font-weight: 600;
+  font-size: 16px;
+  line-height: 150%;
+  color: ${grey[700]};
+  text-transform: capitalize;
+`;
+
+const StepDetail = styled(Box)`
+  margin: 10px 0;
+`;
+
+const DescriptionDetail = styled(Box)`
+  margin: 20px 0;
+`;
+
+const RecipeDetailText = styled(Typography)`
+  color: ${grey[700]};
+  font-weight: 400;
+  font-size: 14px;
+  line-height: 150%;
+  display: flex;
+  align-items: center;
+`;
+
+const StepsGridWrapper = styled(Box)`
+  display: grid;
+  grid-template-columns: 20% 80%;
+  align-items: center;
+  padding: 10px;
+`;
+
+const StepsGridHead = styled(Typography)`
+  font-weight: 600;
+  font-size: 16px;
+  line-height: 150%;
+  color: ${grey[700]};
+`;
+
+const HeadDivider = styled(Box)`
+  width: 100%;
+  border-bottom: 1px solid ${grey[700]};
+  opacity: 0.6;
+`;
+
+const CellDivider = styled(Box)`
+  width: 100%;
+  border-bottom: 1px solid ${grey[300]};
+`;
+
+const VerticalDivider = styled(Box)`
+  width: 1px;
+  height: 25px;
+  border-left: 1px solid ${grey[300]};
+  margin: 0 10px;
+`;
+
+export default function({ recipeDetails, onCloseDetail }: IRecipeDetailsProps) {
+  return (
+    <StyledDrawer open={true} data-testid="select-column-panel" anchor="right">
+      <DrawerContainerBox role="presentation" data-testid="select-column-drawer">
+        <DrawerHeader onCloseDetail={onCloseDetail} />
+        <RecipeDetailWrapper>
+          <RecipeName>{recipeDetails.recipeName}</RecipeName>
+          <StepDetail>
+            <RecipeDetailText>
+              {`${recipeDetails.directives.length} ${T.translate(
+                `${PREFIX}.recipeStepsTableHead.recipeStep`
+              )}`}
+              <VerticalDivider /> {dateFormatting(recipeDetails.createdTimeMillis)}
+            </RecipeDetailText>
+          </StepDetail>
+          <DescriptionDetail>
+            <RecipeDetailText>{recipeDetails.description}</RecipeDetailText>
+          </DescriptionDetail>
+          <StepsGridWrapper>
+            <StepsGridHead>{T.translate(`${PREFIX}.recipeStepsTableHead.serialNo`)}</StepsGridHead>
+            <StepsGridHead>
+              {T.translate(`${PREFIX}.recipeStepsTableHead.recipeStep`)}
+            </StepsGridHead>
+          </StepsGridWrapper>
+          <HeadDivider />
+          {recipeDetails.directives.length > 0 &&
+            recipeDetails.directives.map((recipeStep, recipeStepIndex) => {
```

Comment:
@richadivami 
we might not need the .length check here
the `Array.map()` won't return anything unless there are elements inside
so basically it will skip the loop

plz check for similar unwanted checks and make sure we remove them or add as per need

Reply:
Done